supporter-product-data/src/main/scala/com/gu/services/ZuoraQuerierService.scala

```diff
@@ -0,0 +1,59 @@
+package com.gu.services
+
+import com.gu.conf.ZuoraQuerierConfig
+import com.gu.model.Stage
+import com.gu.model.states.QueryType
+import com.gu.model.states.QueryType.{Full, Incremental}
+import com.gu.model.zuora.request.{BatchQueryRequest, ZoqlExportQuery}
+import com.gu.model.zuora.response.{BatchQueryErrorResponse, BatchQueryResponse}
+import com.gu.okhttp.RequestRunners.FutureHttpClient
+import com.gu.rest.WebServiceHelper
+import io.circe.syntax.EncoderOps
+
+import java.time.{LocalDate, ZoneId, ZoneOffset}
+import java.util.UUID
+import scala.collection.Map.empty
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.classTag
+
+class ZuoraQuerierService(val config: ZuoraQuerierConfig, client: FutureHttpClient)(implicit ec: ExecutionContext)
+  extends WebServiceHelper[BatchQueryErrorResponse] {
+
+  override val wsUrl = config.url
+  override val httpClient: FutureHttpClient = client
+  val authHeaders = Map(
+    "apiSecretAccessKey" -> config.password,
+    "apiAccessKeyId" -> config.username
+  )
+
+  def postQuery(queryType: QueryType): Future[BatchQueryResponse] = {
+    val queries = queryType match {
+      case Full => List(
+        ZoqlExportQuery(
+          s"${SelectActiveRatePlansQuery.name}-${UUID.randomUUID().toString}",
```

Comment:
```suggestion
          s"${SelectActiveRatePlansQuery.name}-${LocalDateTime.now(ZoneId.of("UTC")).toString}",
```
might be a little more helpful for later debugging
```
scala> LocalDateTime.now(ZoneId.of("UTC")).toString
res2: String = 2021-02-05T17:31:10.332
```